data/battle_scripts_1.s

```diff
@@ -1596,27 +1653,10 @@ BattleScript_EffectPartingShotTryAtk:
 BattleScript_EffectPartingShotTrySpAtk:
 	playstatchangeanimation BS_TARGET, BIT_SPATK, STAT_CHANGE_NEGATIVE
 	setstatchanger STAT_SPATK, 1, TRUE
-	statbuffchange STAT_CHANGE_ALLOW_PTR, BattleScript_EffectPartingShotSwitch
+	statbuffchange STAT_CHANGE_ALLOW_PTR, BattleScript_MoveSwitch
 	printfromtable gStatDownStringIds
 	waitmessage B_WAIT_TIME_LONG
-BattleScript_EffectPartingShotSwitch:
-	moveendall
```

Comment:
Notice how you removed the moveendall here but it's no longer present in your new `BattleScript_MoveSwitch`. I don't think you should add it there as not every pivoting attack will need it, but you need to add it back to whichever attacks it was removed from.

Reply:
1. I really appreciate your help and patience with this PR.
2. I restored `BattleScript_EffectPartingShotSwitch` which now calls `moveendall` before jumping to `BattleEffect_MoveSwitch`. I believe that will restore the original logic flow